tests/test_other.py

```diff
@@ -2037,25 +2040,26 @@ def test_m_mm(self):
 
   def test_emcc_debug_files(self):
     if os.environ.get('EMCC_DEBUG'): return self.skip('cannot run in debug mode')
+    canonical_temp_dir = self.canonical_temp_dir()
```

Comment:
this seems confusing. do we need a function there, can't it always be a string?

but a larger question, why is the test runner in charge of declaring the canonical temp dir? shouldn't `shared.py` do that?

Reply:
Hmm, it can be a string. I'll fix it.

This PR dynamically assigns temporary directory path for each test runner thread so that each thread won't get any conflict when writing files in its canonical temp directory. This is not needed for normal emcc use, and `shared.py` is not related to parallel tests, so this is done in test runners.

Reply:
I see. I'm not sure, but I think we can get that benefit (each test is independent) by cleaning up the shared.py temp dir code, and using `EMCC_TEMP_DIR` or one of those. That seems better as the cleanup would be good by itself, and we wouldn't need any new code in the test harness for it.

As I said I'm not sure it can work, and it's a different approach than this PR, so I'd understand if you're not interested to investigate it, I could do that.